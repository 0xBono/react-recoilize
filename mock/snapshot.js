export const filteredCurSnapMock = {
  dummyAtom1: {
    contents: {hello: [], hi: []},
    nodeDeps: [],
    nodeToNodeSubscriptions: [],
    type: 'RecoilState',
  },
  listState: {
    contents: [{text: 'list item'}, {text: 'list item'}, {text: 'list item'}],
    nodeDeps: [],
    nodeToNodeSubscriptions: ['selectorTest', 'stateLengths'],
    type: 'RecoilState',
  },
  listState2: {
    contents: [{text: 'list item'}, {text: 'list item'}, {text: 'list item'}],
    nodeDeps: [],
    nodeToNodeSubscriptions: ['stateLengths'],
    type: 'RecoilState',
  },
  selectorTest: {
    contents: 'test',
    nodeDeps: ['listState'],
    nodeToNodeSubscriptions: [],
    type: 'RecoilValueReadOnly',
  },
  stateLengths: {
    contents: 6,
    nodeDeps: ['listState', 'listState2'],
    nodeToNodeSubscriptions: [],
    type: 'RecoilValueReadOnly',
  },
};
