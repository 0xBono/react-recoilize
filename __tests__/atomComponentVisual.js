import React from 'react';
import AtomComponentVisualContainer from '../src/app/components/ComponentGraph/AtomComponentContainer';
import {
  render,
  fireEvent,
  waitFor,
  screen,
  cleanup,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import '@babel/polyfill';
import {componentAtomTreeMock, filteredCurSnapMock} from '../mock/snapshot.js';

afterEach(cleanup);

xit('renders & matches snapshot - no props', () => {
  const {asFragment} = render(<AtomComponentVisualContainer />);
  expect(asFragment()).toMatchSnapshot();
});

xit('renders & matches snapshot - componetAtomTree props', () => {
  const {asFragment} = render(
    <AtomComponentVisualContainer
      filteredSnapshot={filteredCurSnapMock}
      componentAtomTree={componentAtomTreeMock}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
