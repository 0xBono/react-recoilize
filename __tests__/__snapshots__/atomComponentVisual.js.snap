// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders & matches snapshot - componetAtomTree props 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="AtomComponentVisual"
      >
        <svg
          height="733.33"
          id="canvas"
          width="400"
        >
          <g
            transform="translate(0, 0), scale(1)"
          >
            <g
              fill="none"
              stroke="#646464"
              stroke-width="5"
            >
              <path
                d="M0,0C0,366.665,-800,366.665,-800,733.33"
              />
              <path
                d="M0,0C0,366.665,-400,366.665,-400,733.33"
              />
              <path
                d="M0,0C0,366.665,800,366.665,800,733.33"
              />
              <path
                d="M-400,733.33C-400,1099.9950000000001,-600,1099.9950000000001,-600,1466.66"
              />
              <path
                d="M-400,733.33C-400,1099.9950000000001,-200,1099.9950000000001,-200,1466.66"
              />
              <path
                d="M800,733.33C800,1099.9950000000001,600,1099.9950000000001,600,1466.66"
              />
              <path
                d="M800,733.33C800,1099.9950000000001,1000,1099.9950000000001,1000,1466.66"
              />
              <path
                d="M-200,1466.66C-200,1833.3250000000003,-400,1833.3250000000003,-400,2199.9900000000002"
              />
              <path
                d="M-200,1466.66C-200,1833.3250000000003,0,1833.3250000000003,0,2199.9900000000002"
              />
              <path
                d="M0,2199.9900000000002C0,2566.655,0,2566.655,0,2933.32"
              />
            </g>
            <g
              stroke-linejoin="round"
              stroke-width="1"
            >
              <g
                class="atomNodes"
                transform="translate(0, 0)"
              >
                <text
                  dy=".31em"
                  stroke="#646464"
                  stroke-width="2"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
                <circle
                  fill="gray"
                  r="50"
                />
                <text
                  dy=".31em"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
              </g>
              <g
                class="atomNodes"
                transform="translate(-800, 733.33)"
              >
                <text
                  dy=".31em"
                  stroke="#646464"
                  stroke-width="2"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
                <circle
                  fill="gray"
                  r="50"
                />
                <text
                  dy=".31em"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
              </g>
              <g
                class="atomNodes"
                transform="translate(-400, 733.33)"
              >
                <text
                  dy=".31em"
                  stroke="#646464"
                  stroke-width="2"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
                <circle
                  fill="gray"
                  r="50"
                />
                <text
                  dy=".31em"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
              </g>
              <g
                class="atomNodes"
                transform="translate(800, 733.33)"
              >
                <text
                  dy=".31em"
                  stroke="#646464"
                  stroke-width="2"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
                <circle
                  fill="gray"
                  r="50"
                />
                <text
                  dy=".31em"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
              </g>
              <g
                class="atomNodes"
                transform="translate(-600, 1466.66)"
              >
                <text
                  dy=".31em"
                  stroke="#646464"
                  stroke-width="2"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
                <circle
                  fill="gray"
                  r="50"
                />
                <text
                  dy=".31em"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
              </g>
              <g
                class="atomNodes"
                transform="translate(-200, 1466.66)"
              >
                <text
                  dy=".31em"
                  stroke="#646464"
                  stroke-width="2"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
                <circle
                  fill="gray"
                  r="50"
                />
                <text
                  dy=".31em"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
              </g>
              <g
                class="atomNodes"
                transform="translate(600, 1466.66)"
              >
                <text
                  dy=".31em"
                  stroke="#646464"
                  stroke-width="2"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
                <circle
                  fill="gray"
                  r="50"
                />
                <text
                  dy=".31em"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
              </g>
              <g
                class="atomNodes"
                transform="translate(1000, 1466.66)"
              >
                <text
                  dy=".31em"
                  stroke="#646464"
                  stroke-width="2"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
                <circle
                  fill="gray"
                  r="50"
                />
                <text
                  dy=".31em"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
              </g>
              <g
                class="atomNodes"
                transform="translate(-400, 2199.9900000000002)"
              >
                <text
                  dy=".31em"
                  stroke="#646464"
                  stroke-width="2"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
                <circle
                  fill="gray"
                  r="50"
                />
                <text
                  dy=".31em"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
              </g>
              <g
                class="atomNodes"
                transform="translate(0, 2199.9900000000002)"
              >
                <text
                  dy=".31em"
                  stroke="#646464"
                  stroke-width="2"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
                <circle
                  fill="gray"
                  r="50"
                />
                <text
                  dy=".31em"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
              </g>
              <g
                class="atomNodes"
                transform="translate(0, 2933.32)"
              >
                <text
                  dy=".31em"
                  stroke="#646464"
                  stroke-width="2"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
                <circle
                  fill="gray"
                  r="50"
                />
                <text
                  dy=".31em"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                />
              </g>
            </g>
          </g>
        </svg>
      </div>
    </div>
    <div
      class="AtomSelectorLegend"
    >
      <div>
        <h4>
          Atoms
        </h4>
        <ul>
          <li
            class="atomLi"
          >
            dummyAtom1: {"hello":[],"hi":[]}
          </li>
          <li
            class="atomLi"
          >
            listState: [{"text":"list item"},{"text":"list item"},{"text":"list item"}]
          </li>
          <li
            class="atomLi"
          >
            listState2: [{"text":"list item"},{"text":"list item"},{"text":"list item"}]
          </li>
        </ul>
      </div>
      <div>
        <h4>
          Selectors
        </h4>
        <ul>
          <li
            class="selectorLi"
          >
            selectorTest: "test"
          </li>
          <li
            class="selectorLi"
          >
            stateLengths: 6
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders & matches snapshot - no props 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="AtomComponentVisual"
      >
        <svg
          height="733.33"
          id="canvas"
          width="400"
        >
          <g
            transform="translate(0, 0), scale(1)"
          >
            <g
              fill="none"
              stroke="#646464"
              stroke-width="5"
            />
            <g
              stroke-linejoin="round"
              stroke-width="1"
            >
              <g
                class="atomNodes"
                transform="translate(0, 0)"
              >
                <text
                  dy=".31em"
                  stroke="#646464"
                  stroke-width="2"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                >
                  placeholder
                </text>
                <circle
                  fill="gray"
                  r="50"
                />
                <text
                  dy=".31em"
                  style="font-size: 3rem; fill: white;"
                  text-anchor="end"
                  x="-75"
                >
                  placeholder
                </text>
              </g>
            </g>
          </g>
        </svg>
      </div>
    </div>
    <div
      class="AtomSelectorLegend"
    >
      <div>
        <h4>
          Atoms
        </h4>
        <ul />
      </div>
      <div>
        <h4>
          Selectors
        </h4>
        <ul />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
